scripts/extend_overhang.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+
+### EXTEND pre-miRNA OVERHANG ###
+# 
+# This program modifies if necessary the start and/or end positions of 
+# the pre-miRNA in the annotation file to have an overhang of a minimum 
+# value passed in "extension" in relation to its mature form(s).
+# Annotation entries located at the extrems will only be extended the allowed
+# number of bases by the chromosome start and end positions.
+# 
+# INPUT
+# ---------
+# i: pre-miRNA annotation file in gff3 format
+# extension: Minimum number of basepairs to have as overhang 
+# chr: Chromosome size
+#
+# OUTPUT
+# ---------
+# o: A gff3 file with all the entries having a minimum overhang of the number
+#   specified in the "extension" parameter. 
+#
+
+from argparse import ArgumentParser
+
+
+### ARGUMENTS ###
+
+parser = ArgumentParser(
+    description="Script to extend pre-miRNs overhang."
+    )
+parser.add_argument(
+    '-v','--version', 
+    action='version', 
+    version='%(prog)s 1.0',
+    help="Show program's version number and exit"
+    )
+parser.add_argument(
+    '-i', '--input', 
+    help="Path to the gff3 input file",
+    type=str
+    )
+parser.add_argument(
+    '-o', '--output', 
+    help="Path for the gff3 final file.",
+    type=str
+    )
+parser.add_argument(
+    '-e','--extension',
+    help="Minimum number of base pairs each overhang must be of. Maximum value = 10. Default value = 10.",
+    default=10,
+    type=int,
+    choices=range(0,11)
+    )
+parser.add_argument(
+    '--chr',
+    help="Path to the file containing the chromosome and its size.",
+    type=str
+    )
+    
+args = parser.parse_args()
+
+with open(args.chr, "r") as f:
+    chr_size = int(f.readline().split()[1])
+
+extension =  args.extension
+
+
+def extend_overhang(pre, mirna_start, mirna_end, extension, chr_size):
+    '''
+    EXTEND OVERHANG FUNCTION
```

Comment:
Please use Google-style docstrings, and make sure to provide them for every module, class, method and function.

Also, please provide type hints at least for all function and method signatures, as well as for global variables (which you should anyway avoid, if at all possible). Also consider adding them for local variables, because, why not. Better to be safe.